node/pkg/accountant/watcher.go

```diff
@@ -196,3 +215,49 @@ func (acct *Accountant) processPendingTransfer(xfer *WasmTransfer) {
 		acct.logger.Info("acctwatch: unknown transfer has been approved, ignoring it", zap.String("msgId", msgId))
 	}
 }
+
+// WasmObservationError represents an error event from the smart contract.
+type (
+	WasmObservationError struct {
+		Key   ObservationKey
+		Error string `json:"error"`
+	}
+)
+
+// parseWasmObservationError parses error events from the smart contract. All other event types are ignored.
+func parseWasmObservationError(logger *zap.Logger, event tmAbci.Event, contractAddress string) (*WasmObservationError, error) {
```

Comment:
Rather than copy-pasting the parse function with minor changes we can make it generic:

```go
func parseEvent[T any](logger zap.Logger, event abci.Event, name, contractAddress string) (T, error) {
	var out T

	if event.Type != name {
		return out, fmt.Errorf(...)
	}
	
	// Convert event attributes into json as usual.
	
	if err := json.Unmarshal(attrBytes, &out); err != nil {
		return out, fmt.Errorf("failed to unmarshal %T from event: %w", out, err)
	}
	return out, nil
}
```

We could also move the `event.Type` check out of the function to avoid passing in the event name as a parameter.